packages/netlify-cms-core/src/actions/config.js

```diff
@@ -31,6 +31,29 @@ const setDefaultPublicFolder = map => {
   return map;
 };
 
+const setSnakeCaseConfig = field => {
+  // List of options that are still using camelCase
+  const widgetKeyMap = new Map([
+    ['date_format', 'dateFormat'],
+    ['time_format', 'timeFormat'],
+    ['picker_utc', 'pickerUtc'],
+    ['editor_components', 'editorComponents'],
+    ['value_type', 'valueType'],
+    ['value_field', 'valueField'],
+    ['search_fields', 'searchFields'],
+    ['display_fields', 'displayFields'],
+    ['options_length', 'optionsLength'],
+  ]);
+
+  widgetKeyMap.forEach((value, key) => {
+    if (field.has(value)) {
+      field.set(key, field.get(value));
+      delete field[value];
```

Comment:
This doesn't work for [Immutable](https://immutable-js.github.io/immutable-js/) (you'd need `field = field.delete(key)`).

Regardless we don't want to delete the old casing (yet), since users might try to access those values in custom widgets (that are not part of this repo).